oximeter/db/src/oxql/plan/plan.rs

```diff
@@ -0,0 +1,1366 @@
+                processed_nodes.push_front(current_node);
+                processed_nodes.push_front(next_node);
+                continue;
+            };
+
+            match next_node {
+                Node::Subquery(subplans) => {
+                    // Push the limit onto the subquery plans and recurse.
+                    let new_subplans = subplans
+                        .into_iter()
+                        .map(|mut plan| {
+                            let start = Instant::now();
+                            let nodes = [
+                                plan.optimized.nodes(),
+                                &[Node::Limit(limit.clone())],
+                            ]
+                            .concat();
+                            plan.optimized = Self::pushdown_limit(&nodes)?;
+                            plan.duration += start.elapsed();
+                            Ok(plan)
+                        })
+                        .collect::<anyhow::Result<Vec<_>>>()?;
+                    processed_nodes.push_front(Node::Subquery(new_subplans));
+                    modified = true;
+                }
+                Node::Get(mut get) => {
+                    // If we've gotten here, we _may_ be able to push this into
+                    // the databse, but only if there isn't one already in the
+                    // get node.
+                    match get.limit.as_mut() {
+                        Some(existing) => {
+                            // There's already a limiting operation here. We may
+                            // be able to coalesce them, but only if they're the
+                            // same kind. If not, it's not correct to reorder
+                            // them, so we have to push the current node and
+                            // then the next (get) node onto the processed list.
+                            if existing.kind == limit.limit.kind {
+                                // Take the smaller of the two!
+                                existing.count =
+                                    existing.count.min(limit.limit.count);
+                                modified = true;
+                            } else {
+                                // These are different kinds, push them in
+                                // order, starting with the limit.
+                                processed_nodes.push_front(Node::Limit(limit));
+                            }
+                        }
+                        None => {
+                            let old = get.limit.replace(limit.limit);
+                            assert!(old.is_none());
+                            modified = true;
+                        }
+                    }
+
+                    // We always push the get node last.
+                    processed_nodes.push_front(Node::Get(get));
+                }
+                Node::Delta(_)
+                | Node::Align(_)
```

Comment:
I don't believe it's correct to push a limit through an alignment, because the latter can change the number of elements. Let's add a test for this specific case, too.